extensions/lifecycle/tasks/LifecycleColdStatusArchiveTask.js

```diff
@@ -0,0 +1,147 @@
+    _executeDeleteData(entry, objMD, log, done) {
+        log.debug('action execution starts', entry.getLogInfo());
+        const { bucketName, objectKey, accountId } = entry.getTarget();
+        const backbeatClient = this.getBackbeatClient(accountId);
+
+        if (!backbeatClient) {
+            log.error('failed to get backbeat client', { accountId });
+            return done(errors.InternalError
+                .customizeDescription('Unable to obtain client'));
+        }
+
+        const locations = objMD.getLocation();
+        const req = backbeatClient.batchDelete({
+            Locations: locations.map(location => ({
+                key: location.key,
+                dataStoreName: location.dataStoreName,
+                size: location.size,
+                dataStoreVersionId: location.dataStoreVersionId,
+            })),
+            Bucket: bucketName,
+            Key: objectKey,
+            StorageClass: objMD.getDataStoreName(),
+            Tags: JSON.stringify({
+                'scal-delete-marker': 'true',
+                'scal-delete-service': 'lifecycle-transition',
+            }),
+        });
+        return req.send(err => {
+            log.info('action execution ended', entry.getLogInfo());
+
+            if (err && err.statusCode === 404) {
+                log.info('unable to find data to delete',
+                    Object.assign({
+                        method: 'LifecycleColdStatusArchiveTask._executeDeleteData',
+                        bucket: bucketName,
+                        key: objectKey,
+                    }, entry.getLogInfo));
+                return done();
+            }
+
+            if (err) {
+                log.error('an error occurred on deleteData method to backbeat route',
+                    Object.assign({
+                        method: 'LifecycleColdStatusArchiveTask._executeDeleteData',
+                        error: err.message,
+                        httpStatus: err.statusCode,
+                    }, entry.getLogInfo()));
+                return done(err);
+            }
+            return done();
+        });
+    }
+
+    processEntry(entry, done) {
+        const log = this.logger.newRequestLogger();
+        let objectMD;
+        return async.series([
+            next => this._getMetadata(entry, log, (err, res) => {
+                if (err) {
+                    return next(err);
+                }
+
+                objectMD = res;
+                return next();
+            }),
+            next => {
+                const archiveInfo = entry.getArchiveInfo();
+                // set new ObjectMDArchive to ObjectMD
+                objectMD.setArchive(new ObjectMDArchive(archiveInfo));
+                this._putMetadata(entry, objectMD, log, next);
+            },
+            next => {
+                const locationToGC = objectMD.getLocation();
+                if (!locationToGC || (Array.isArray(locationToGC) && locationToGC.length === 0)) {
```

Comment:
I don't think the second half of this condition can be true if the first half is not already true since an empty array is also falsy.